stixcore/io/ProcessingHistoryStorage.py

```diff
@@ -0,0 +1,212 @@
+import sqlite3
+from pathlib import Path
+from datetime import datetime
+
+from stixcore.util.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class ProcessingHistoryStorage:
+    """Persistent handler for meta data on already processed Products"""
+
+    def __init__(self, filename):
+        """Create a new persistent handler. Will open or create the given sqlite DB file.
+
+        Parameters
+        ----------
+        filename : path like object
+            path to the sqlite database file
+        """
+        self.conn = None
+        self.cur = None
+        self.filename = filename
+        self._connect_database()
+
+    def _connect_database(self):
+        """Connects to the sqlite file or creates an empty one if not present."""
+        try:
+            self.conn = sqlite3.connect(self.filename)
+            self.cur = self.conn.cursor()
+            logger.info('ProcessingHistoryStorage DB loaded from {}'.format(self.filename))
+
+            # TODO reactivate later
+            # self.cur.execute('''CREATE TABLE if not exists processed_flare_products (
+            #             id INTEGER PRIMARY KEY AUTOINCREMENT,
+            #             flareid TEXT NOT NULL,
+            #             flarelist TEXT NOT NULL,
+            #             version INTEGER NOT NULL,
+            #             name TEXT NOT NULL,
+            #             level TEXT NOT NULL,
+            #             type TEXT NOT NULL,
+            #             fitspath TEXT NOT NULL,
+            #             p_date FLOAT NOT NULL
+            #         )
+            # ''')
+            # self.cur.execute('''CREATE INDEX if not exists processed_flare_products_idx ON
+            #                     processed_flare_products (flareid, flarelist, version, name,
+            # level, type)''')
+
```

Comment:
Does this need to be uncommented?

Reply:
yes as mentioned i removed/uncomment some parts that not yet are necessary but already be prepared for images on flarelist entries